exchanges/btcmarkets/btcmarkets.go

```diff
@@ -52,316 +75,472 @@ type BTCMarkets struct {
+	var resp TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsWithdrawals+"/"+id,
+		nil, &resp)
 }
 
-// WithdrawCrypto withdraws cryptocurrency into a designated address
-func (b *BTCMarkets) WithdrawCrypto(amount float64, currency, address string) (string, error) {
-	newAmount := int64(amount * float64(common.SatoshisPerBTC))
+// ListDeposits lists the deposit history
+func (b *BTCMarkets) ListDeposits() ([]TransferData, error) {
+	var resp []TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsDeposits, nil, &resp)
+}
 
-	req := WithdrawRequestCrypto{
-		Amount:   newAmount,
-		Currency: strings.ToUpper(currency),
-		Address:  address,
-	}
+// GetDeposit gets deposit info for a given ID
+func (b *BTCMarkets) GetDeposit(id string) (TransferData, error) {
+	var resp TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsDeposits+"/"+id,
+		nil, &resp)
+}
 
-	resp := Response{}
-	err := b.SendAuthenticatedRequest(http.MethodPost,
-		btcMarketsWithdrawCrypto,
-		req,
+// ListTransfers lists the past asset transfers
+func (b *BTCMarkets) ListTransfers() ([]TransferData, error) {
+	var resp []TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsTransfers, nil, &resp)
+}
+
+// GetTransfer gets asset transfer info for a given ID
+func (b *BTCMarkets) GetTransfer(id string) (TransferData, error) {
+	var resp TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsTransfers+"/"+id,
+		nil, &resp)
+}
+
+// FetchDepositAddress gets deposit address for the given asset
+func (b *BTCMarkets) FetchDepositAddress(assetName string) (DepositAddress, error) {
+	var resp DepositAddress
+	params := url.Values{}
+	params.Set("assetName", assetName)
+	return resp, b.SendAuthenticatedRequest(http.MethodGet,
+		common.EncodeURLValues(btcMarketsAddresses, params),
+		nil,
 		&resp)
-	if err != nil {
-		return "", err
-	}
+}
 
-	if !resp.Success {
-		return "", errors.New(resp.ErrorMessage)
-	}
+// GetWithdrawalFees gets withdrawal fees for all assets
+func (b *BTCMarkets) GetWithdrawalFees() ([]WithdrawalFeeData, error) {
+	var resp []WithdrawalFeeData
+	return resp, b.SendHTTPRequest(btcMarketsAPIURL+btcMarketsAPIVersion+btcMarketsWithdrawalFees,
+		&resp)
+}
 
-	return resp.Status, nil
+// ListAssets lists all available assets
+func (b *BTCMarkets) ListAssets() ([]AssetData, error) {
+	var resp []AssetData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsAssets, nil, &resp)
 }
 
-// WithdrawAUD withdraws AUD into a designated bank address
-// Does not return a TxID!
-func (b *BTCMarkets) WithdrawAUD(accountName, accountNumber, bankName, bsbNumber string, amount float64) (string, error) {
-	newAmount := int64(amount * float64(common.SatoshisPerBTC))
+// GetTransactions gets trading fees
+func (b *BTCMarkets) GetTransactions(assetName string) ([]TransactionData, error) {
```

Comment:
assetName is an optional parameter
This does not seem to support optional parameters:
https://api.btcmarkets.net/doc/v3#tag/Account-APIs/paths/~1v3~1accounts~1me~1transactions/get

Reply:
Please address previous nit on this